Classes/ios/Scanners/MTBBarcodeScanner.m

```diff
@@ -561,57 +566,60 @@ - (void)setDeviceInput:(AVCaptureDeviceInput *)deviceInput session:(AVCaptureSes
     [self removeDeviceInput];
     
     self.currentCaptureDeviceInput = deviceInput;
-    
-    if ([deviceInput.device lockForConfiguration:nil] == YES) {
-        
-        // Prioritize the focus on objects near to the device
-        if ([deviceInput.device respondsToSelector:@selector(isAutoFocusRangeRestrictionSupported)] &&
-            deviceInput.device.isAutoFocusRangeRestrictionSupported) {
-            
-            self.initialAutoFocusRangeRestriction = deviceInput.device.autoFocusRangeRestriction;
-            deviceInput.device.autoFocusRangeRestriction = AVCaptureAutoFocusRangeRestrictionNear;
-        }
-        
-        // Focus on the center of the image
-        if ([deviceInput.device respondsToSelector:@selector(isFocusPointOfInterestSupported)] &&
-            deviceInput.device.isFocusPointOfInterestSupported) {
-            
-            self.initialFocusPoint = deviceInput.device.focusPointOfInterest;
-            deviceInput.device.focusPointOfInterest = CGPointMake(kFocalPointOfInterestX, kFocalPointOfInterestY);
-        }
-        
-        [self updateTorchModeForCurrentSettings];
-        
-        [deviceInput.device unlockForConfiguration];
-    }
-    
+    [self updateFocusPreferencesOfInput:deviceInput reset:NO];
+
     [session addInput:deviceInput];
 }
 
 - (void)removeDeviceInput {
     AVCaptureDeviceInput *deviceInput = self.currentCaptureDeviceInput;
+
     if (deviceInput == nil) {
         // No need to remove the device input if it was never set
         return;
     }
     
     // Restore focus settings to the previously saved state
-    if ([deviceInput.device lockForConfiguration:nil] == YES) {
-        if ([deviceInput.device respondsToSelector:@selector(isAutoFocusRangeRestrictionSupported)] &&
-            deviceInput.device.isAutoFocusRangeRestrictionSupported) {
-            deviceInput.device.autoFocusRangeRestriction = self.initialAutoFocusRangeRestriction;
+    [self updateFocusPreferencesOfInput:deviceInput reset:YES];
+    
+    [self.session removeInput:deviceInput];
+    self.currentCaptureDeviceInput = nil;
+}
+
+- (void)updateFocusPreferencesOfInput:(AVCaptureDeviceInput *)deviceInput reset:(BOOL)reset {
+    NSParameterAssert(deviceInput);
+
+    if (!deviceInput) {
+        return;
+    }
+
+    NSError *lockError;
+
+    if ([deviceInput.device lockForConfiguration:&lockError]) {
+        // Prioritize the focus on objects near to the device
+        if (deviceInput.device.isAutoFocusRangeRestrictionSupported) {
+            if (!reset) {
+                self.initialAutoFocusRangeRestriction = deviceInput.device.autoFocusRangeRestriction;
+            }
```

Comment:
Yes. Also changed the parameter type to hand over the device directly so we don't call `deviceInput.device` on every line.